runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/BoundedQueueExecutor.java

```diff
@@ -149,18 +191,22 @@ public String summaryHtml() {
       builder.append(executor.getMaximumPoolSize());
       builder.append("<br>/n");
 
+      builder.append("Maximum Threads: ");
+      builder.append(maximumThreadCount());
+      builder.append("<br>/n");
+
       builder.append("Active Threads: ");
       builder.append(executor.getActiveCount());
       builder.append("<br>/n");
 
       builder.append("Work Queue Size: ");
-      builder.append(elementsOutstanding);
+      builder.append(elementsOutstanding());
```

Comment:
should the monitor.enter() at the top of summaryHtml be removed if using the accessors that grab the monitor?

not sure if it is reentrant. Would be good to add renderHtml to the unit test either way.

Reply:
Ah I didn't notice that. Reverted and added a test case for it.